Libraries/Microsoft.Teams.Plugins/Microsoft.Teams.Plugins.AspNetCore/AspNetCorePlugin.cs

```diff
@@ -139,4 +147,85 @@ await Events(
             return new Response(System.Net.HttpStatusCode.InternalServerError, ex.ToString());
         }
     }
+
+    public async Task<IResult> Do(HttpContext httpContext, CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            var request = httpContext.Request;
+            var token = ExtractToken(request);
+            var activity = await ExtractActivity(request);
+
+            if (activity is null)
+            {
+                return Results.BadRequest("Missing activity");
+            }
+
+            var data = new Dictionary<string, object?>
+            {
+                ["Request.TraceId"] = httpContext.TraceIdentifier
+            };
+
+            foreach (var pair in httpContext.Items)
+            {
+                var key = pair.Key.ToString();
+
+                if (key is null) continue;
+
+                data[key] = pair.Value;
+            }
```

Comment:
hmm, we don't do this in other SDKs. We should right? What is essential for propagation @singhk97 ?

Reply:
I believe I originally did this in the controller to support a PA ask about adding their own context data in a key value pair, this comes off the request so if node/python have a concept like this we should def carry it over. I don't believe express does (may be mistaken) but the JS implementation of the context is already setup to access key/value pairs.